lib/auth/join_oracle_test.go

```diff
@@ -0,0 +1,445 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package auth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+	"github.com/oracle/oci-go-sdk/v65/common"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth/join/oracle"
+	"github.com/gravitational/teleport/lib/auth/testauthority"
+	"github.com/gravitational/teleport/lib/fixtures"
+)
+
+func TestCheckHeaders(t *testing.T) {
+	t.Parallel()
+	clock := clockwork.NewFakeClock()
+
+	defaultChallenge := "abcd1234"
+	defaultAuthHeader := fmt.Sprintf(`Signature headers="%s %s"`, oracle.DateHeader, oracle.ChallengeHeader)
+
+	t.Run("ok", func(t *testing.T) {
+		headers := formatHeaderFromMap(map[string]string{
+			"Authorization":        defaultAuthHeader,
+			oracle.DateHeader:      clock.Now().UTC().Format(http.TimeFormat),
+			oracle.ChallengeHeader: defaultChallenge,
+		})
+		require.NoError(t, checkHeaders(headers, defaultChallenge, clock))
+	})
+
+	tests := []struct {
+		name    string
+		headers map[string]string
+	}{
+		{
+			name: "missing signed headers",
+			headers: map[string]string{
+				"Authorization":        `Signature foo="bar"`,
+				oracle.DateHeader:      clock.Now().UTC().Format(http.TimeFormat),
+				oracle.ChallengeHeader: defaultChallenge,
+			},
+		},
+		{
+			name: "date not signed",
+			headers: map[string]string{
+				"Authorization":        fmt.Sprintf(`Signature headers="%s"`, oracle.ChallengeHeader),
```

Comment:
Optional: since we are testing this with data that would come from an external service, is supposed to meet external contract, and is just a short string, consider abandoning constants here, and instead, explicitly specifying how the externally-provided text looks like.